app/src/main/java/com/kickstarter/ui/adapters/RewardAndAddOnsAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package com.kickstarter.ui.adapters
+
+import android.util.Pair
+import android.view.View
+import com.kickstarter.R
+import com.kickstarter.models.Reward
+import com.kickstarter.ui.data.ProjectData
+import com.kickstarter.ui.viewholders.*
+
+class RewardAndAddOnsAdapter(private val viewListener: RewardAndAddOnsAdapter.ViewListener) : KSAdapter() {
+
+    interface ViewListener:RewardViewHolder.Delegate, AddOnViewHolder.ViewListener
```

Comment:
It looks like both of these interfaces do the same thing (listen for user input), I found Delegate to be confusing at first. ViewListener makes sense. Is that why you changed it?

Reply:
For now it's empty and doing the same thing because I've implemented the same "action" as the rewards, that's quite sure gonna change. I just what to handle both view listener one for rewards and the other for AddOns, I didn't really change the name because AddOns are in essence  rewards ... anyway, not sure yet how to call that action, any suggestion is welcome :) 
And the Delegate part comes from RewardViewHolder, which is already implemented, the naming is because that code was started by iOS developers and that's how the call the viewListeners 😏